workers/unity/Packages/com.improbable.gdk.core/Collections/CollectionProvider.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+
+namespace Improbable.Gdk.Core.Collections
+{
+    internal static class CollectionProvider<TProvider, TValue> where TProvider : ICollection<TValue>, new()
```

Comment:
Should this be called TProvider if it's a collection?

Reply:
👍